### PR TITLE
Fix error navigating sections

### LINKED ISSUE #1379: Navigating through sections using navigation results in 5xx's
### Expected behaviour
navigate forward to sections not visited yet should be fine

### Actual behaviour
when navigating to a different section, user is presented with a 5xx

### Steps to reproduce the behaviour
load any survey with multiple sections and navigation, eg Census or MWSS

### Technical information
`TypeError` thrown

#### Browser
all 

### Screenshot
screen shot 2017-11-22 at 08 29 58

### PR DESCRIPTION
**Pull request by jonnyshaw89**
### What is the context of this PR?
The first block of each section was not being added to the routing path if a group before got to a dead end in the routing.

Fixes https://github.com/ONSdigital/eq-survey-runner/issues/1379

### How to review 
Does it fix the issue?

### CODE REVIEW

**Review by dwyeradam — CHANGES_REQUESTED** [review]
The group routing in test_routing_group.json no longer works.

**Review by shahi645 — CHANGES_REQUESTED** [review]
Still getting 5xx's, i think we should test generated sections as thats what sees to cause the issue

**Comment by jonnyshaw89** [comment]
@shahi645 i'm not sure what you mean. Can you give a detailed list of steps to reproduce?

File: tests/functional/spec/navigation.spec.js [hunk 1]
```diff
@@ -151,6 +151,15 @@ describe('Navigation', function() {
       });
   });
 
+  it('Given I have a section that can not be reached from the start of the survey, When I click on the navigation link,' +
+    ' Then the first block of that section should be displayed', function() {
+    return helpers.openQuestionnaire('test_navigation_routing.json').then(() => {
```
Comment (MIslam92) [inline tests/functional/spec/navigation.spec.js:157]: Would it not be better to just do a beforeEach to open the questionnaire rather then having to do it for every scenario for the Navigation?

**Review by MIslam92 — DISMISSED** [review]

File: tests/functional/spec/navigation.spec.js (hunk 1 above)
Comment (jonnyshaw89) [reply tests/functional/spec/navigation.spec.js:157]: @MIslam92 this questionnaire is only used once.

Comment (MIslam92) [reply tests/functional/spec/navigation.spec.js:157]: @jonnyshaw89 Sorry, I was referring to when the test_navigation.json was used my bad

**Comment by shahi645** [comment]
@jonnyshaw89 
- Load MWSS 
- Select several pay patterns
- Try navigating to a later section via navigation, and you should see a 500

**Comment by shahi645** [comment]
@jonnyshaw89 Tested out just now, the changes are working. Cant reproduce error 👍

**Review by shahi645 — APPROVED** [review]
After a thorough test this is working as it should now, affected surveys work fine

**Review by dwyeradam — APPROVED** [review]

**Comment by dwyeradam** [comment]
Have added an issue #1386 we found during this PR. Using routing to skip Sections should not be allowed at schema validation.

**Review by dwyeradam — CHANGES_REQUESTED** [review]
Do we have a test that would currently fail on master?

**Review by dwyeradam — APPROVED** [review]

**Pull request merged**